utils.go

```diff
@@ -16,18 +17,20 @@ const (
 	TagsHTTPHeaderPrefix = "Open-Tracing-Trace-Tags-"
 )
 
-// PropagateSpanInHeader encodes Span `ctx` to `h` as a series of
-// HTTP headers. Values are URL-escaped.
-func PropagateSpanInHeader(
+// InjectSpanInHeader encodes Span `sp` in `h` as a series of HTTP headers.
+// Values are URL-escaped.
+func InjectSpanInHeader(
 	sp Span,
 	h http.Header,
-	propagator SpanPropagator,
-) {
-	contextIDMap, tagsMap := propagator.PropagateSpanAsText(sp)
-	for headerSuffix, val := range contextIDMap {
+) error {
+	var traceState, attrs map[string]string
+	if err := InjectSpan(sp, WIRE_ENCODING_SPLIT_TEXT, &traceState, &attrs); err != nil {
```

Comment:
... just a simple example of how injection would look for the common-case of HTTP headers. See the extraction complement below.